.github/workflows/main.yml

```diff
@@ -1,61 +1,92 @@
 name: 🚀
 
 on:
-  pull_request: { }
-  push: { branches: [ main ] }
+  pull_request: {}
+  push: { branches: [main] }
 
 jobs:
   test:
     name: 🧪 Test
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
-      - uses: actions/setup-node@v3
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
         with:
-          node-version: 16
+          node-version: 20
           cache: 'yarn'
       - name: install dependencies
         run: yarn && yarn --cwd ./functions
       - name: run linter
         run: yarn lint && yarn --cwd ./functions lint
       - name: run tests
         run: yarn test && yarn --cwd ./functions test
-#      - name: Upload test results
-#        uses: actions/upload-artifact@v1
-#        with:
-#          name: Test results
-#          path: "**/artifacs"
+  #      - name: Upload test results
+  #        uses: actions/upload-artifact@v1
+  #        with:
+  #          name: Test results
+  #          path: "**/artifacs"
 
   build:
     name: 🛠 Build
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v3
-      - uses: actions/setup-node@v3
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
         with:
-          node-version: 16
+          node-version: 20
           cache: 'yarn'
       - name: install dependencies
         run: yarn && yarn --cwd ./functions
       - name: build
         run: yarn --cwd ./functions build
 
+  testDeploy:
+    name: Test Deploy
+    needs: [test, build]
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 20
+          cache: 'yarn'
+      - name: install dependencies
+        run: yarn && yarn --cwd ./functions
+      - name: Deploy test to Firebase
+        uses: w9jds/firebase-action@v13.10.2
+        with:
+          args: deploy --only functions:testFunction
+        env:
+          GCP_SA_KEY: '${{ secrets.FIREBASE_SERVICE_ACCOUNT_UNITY_CI_VERSIONS }}'
+      - name: Cleanup Firebase Test
+        uses: w9jds/firebase-action@v13.10.2
+        with:
+          args: functions:delete testFunction --force
+        env:
+          GCP_SA_KEY: '${{ secrets.FIREBASE_SERVICE_ACCOUNT_UNITY_CI_VERSIONS }}'
```

Comment:
Do we still need this?

Reply:
I would want to keep it as this is a great way to ensure the project will deploy properly for the actual functions since it requires roughly the same permissions as the main functions and has been setup to require all the secrets that the other functions might use as a deploy test